### DISCUSSION TITLE
React, Formulários e consumo de APIs

### DISCUSSION THREAD

**Discussion by matheusrod92** [frontendbr/forum#1065] [Perguntas e Respostas]
Fala galera beleza? Estamos desacoplando um monolito aqui da empresa e estou experimentando react como framework para o desenvolvimento da UI mas estou com varias duvidas.

Primeiro gostaria de saber qual a estrutura de API vocês utilizam (JSON Schema, JSON API, GraphQL, outros) e porque?

E sobre a dúvida é o seguinte, tenho um formulário de feedbacks para persistir o dado eu envio um **POST** com o seguinte objeto:
```
system_feedback: {
  score: 10,
  observations: "foobar"
}
```
O dado é persistido normalmente mas aqui começa o meu problema porque o feedback é unico por usuário e eu não posso enviar outro post, então pensei em fazer uma requisição **GET** no `componentDidMount()` para verificar se o formulário já foi submetido (faz sentido isso?) salvando em uma variável no meu state.
Caso o form já tenha sido preenchido (verificação através do state) eu altero o verbo de **POST** para **PATCH/PUT** que neste caso funcionaria mas não vejo como uma alternativa "escalavel".

Por favor indiquem libs que facilitem manipulação de forms e apis hahahahah

Valeu

**EDIT:**
Essa rota da nossa API é aberta apenas para os verbos: **POST** e **PATCH** mas eu dei um exemplo caso ela estivesse aberta para o verbo **GET** só para verificar se o form já foi submetido. Mas como não temos essa alternativa então de opiniões sobre este caso também!

**Comment by ninetails** [comment]
Sobre estrutura de API, utilizei até agora somente REST. Geralmente chamo a API no `componentDidMount`, seja por fetch ou dispatch pro Redux que faz o fetch.

Podendo escolher, nunca utilizei mas diria que o melhor das alternativas seria o GraphQL pois vc poderia sincronizar o dado com o estado que vc tem na sua aplicação.

Sobre a dúvida, sim, fazer um **GET** faz bastante sentido, perfeitamente como vc diagnosticou (hidratar seu estado com o dado do formulário enviado), ainda mais pensando em melhorar o desacoplamento.
